File: core/src/com/shadyaardvark/geom/Orientation.java
package com.shadyaardvark.geom;

public enum Orientation {
    POINTY(Math.sqrt(3.0), Math.sqrt(3.0) / 2.0, 0.0, 3.0 / 2.0, Math.sqrt(3.0) / 3.0, -1.0 / 3.0, 0.0, 2.0 / 3.0, 0.5),
    FLAT(3.0 / 2.0, 0.0, Math.sqrt(3.0) / 2.0, Math.sqrt(3.0), 2.0 / 3.0, 0.0, -1.0 / 3.0, Math.sqrt(3.0) / 3.0, 0.0);

    final double f0;
    final double f1;
    final double f2;
    final double f3;
    final double b0;
    final double b1;
    final double b2;
    final double b3;
    final double start_angle;

    Orientation(double f0, double f1, double f2, double f3, double b0, double b1, double b2, double b3, double start_angle) {
        this.f0 = f0;
        this.f1 = f1;
        this.f2 = f2;
        this.f3 = f3;
        this.b0 = b0;
        this.b1 = b1;
        this.b2 = b2;
        this.b3 = b3;
        this.start_angle = start_angle;
    }
}

File: core/src/com/trashcob/dicewars/math/geom/hexagon/Orientation.kt
package com.trashcob.dicewars.math.geom.hexagon

enum class Orientation(val f0: Float, val f1: Float, val f2: Float, val f3: Float,
                       val b0: Float, val b1: Float, val b2: Float, val b3: Float,
                       val startAngle: Float) {
    POINTY(Math.sqrt(3.0).toFloat(), Math.sqrt(3.0).toFloat() / 2f, 0f, 3f / 2f,
            Math.sqrt(3.0).toFloat() / 3f, -1f / 3f, 0f, 2f / 3f,
            0.5f),
    FLAT(3f / 2f, 0f, Math.sqrt(3.0).toFloat() / 2f, Math.sqrt(3.0).toFloat(),
            2f / 3f, 0f, -1f / 3f, Math.sqrt(3.0).toFloat() / 3f,
            0f)
}
